tools/hicup/hicup_digester.xml

```diff
@@ -0,0 +1,53 @@
+<tool id="hicup_digester" name="Digester" version="0.1.0">
+    <requirements>
+        <requirement type="package" version="0.5.9">hicup</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:" />
+    </stdio>
+    <command><![CDATA[
+        hicup_digester --re1 $re1 --genome $genome
+        #if $advanced_options.re2:
+            --re2 $advanced_options.re2
+        #end if
+        $advanced_options.zip 
+        #for $file in $inputFiles:
+            $file
+        #end for
+    ]]></command>
+    <inputs>
+    <param name="inputFiles" type="data" multiple="true" format="fa" label="Input DNA sequence files that should be digested"/>
+    <param argument="--re1" type="text" label="Restriction enzyme" help="Restriction enzyme used to digest the genome (the enzyme that 
+            forms the ligation junction) e.g. A^GATCT,BglII.  Some Hi-C protocols may use two enzymes at this stage.  To specify two enzymes: -1 A^GATCT,BglII:A^AGCTT,HindIII."/>
+    <param argument="--genome" type="text" label="Genome" help="Name of the genome to be digested (not the path to the genome 
+            file or files, but the genome name to include in the output file)"/>
+    <section name="advanced_options" title="Advanced options">
+        <param argument="--re2" type="text" label="Restriction enzyme instead of sonication to shorten di-tags." help="To specify a restriction enzyme instead of sonication to
+                shorten di-tags. This restriction site does NOT form a Hi-C ligation junction. 2 .g. AG^CT,AluI. Typically the sonication protocol is followed."/>
+        <param argument="--zip" type="boolean" value="false" truevalue="--zip" falsevalue="" label="Compress output" help="Compress the output with zip"/>
+    </section>
+    </inputs>
+    <outputs>
+        <data format="txt" name="digest_file">
+            <discover_datasets pattern="__designation_and_ext__" visible="true" />
```

Comment:
We spoke about this issue before the Christmas break. HiCUP produces many output files and it is way easier to detect instead of hardcoding them. Additional, many file names have time stamps and or depend on the input files names but of course only partially. I can do all by hand and hardcode it but this a lot more sensitive to errors.